qupath-gui-fx/src/main/java/qupath/lib/gui/charts/ScatterPlotChart.java

```diff
@@ -0,0 +1,337 @@
+            getYAxis().setTickLabelsVisible(newV);
+        });
+        maxPoints.addListener((obs, oldV, newV) -> {
+            if (newV.intValue() < 0) {
+                maxPoints.set(0);
+            }
+            resampleAndUpdate();
+        });
+        xLabel.addListener((obs, oldV, newV) -> {
+            getXAxis().setLabel(newV);
+        });
+        yLabel.addListener((obs, oldV, newV) -> {
+            getYAxis().setLabel(newV);
+        });
+        rngSeed.addListener((obs, oldV, newV) -> resampleAndUpdate());
+        setHorizontalGridLinesVisible(drawGrid.get());
+        setVerticalGridLinesVisible(drawGrid.get());
+        getXAxis().setTickLabelsVisible(drawGrid.get());
+        getYAxis().setTickLabelsVisible(drawAxes.get());
+
+        setAnimated(false);
+
+        series.add(new Series<>());
+        setData(series);
+
+        resampleAndUpdate();
+    }
+
+    /**
+     * Set the maximum number of points that the plot will display.
+     * Subsampling will be used to ensure only this number of points or fewer are displayed.
+     * @param maxPoints The maximum number of elements
+     */
+    public void setMaxPoints(int maxPoints) {
+        this.maxPoints.set(maxPoints);
+    }
+
+    /**
+     * Set whether gridlines are drawn on the plot.
+     * @param drawGrid Whether gridlines are drawn on the plot.
+     */
+    public void setDrawGrid(boolean drawGrid) {
+        this.drawGrid.set(drawGrid);
+    }
+
+    /**
+     * Set whether gridlines are drawn on the plot.
+     * @param drawAxes Whether axis ticks are drawn on the plot.
+     */
+    public void setDrawAxes(boolean drawAxes) {
+        this.drawAxes.set(drawAxes);
+    }
+
+    /**
+     * Set the RNG seed for subsampling
+     * @param rngSeed The random number generator seed
+     */
+    public void setRNG(int rngSeed) {
+        this.rngSeed.set(rngSeed);
+    }
+
+    /**
+     * Set point opacity
+     * @param pointOpacity the point opacity
+     */
+    public void setPointOpacity(double pointOpacity) {
+        this.pointOpacity.set(pointOpacity);
+    }
+
+    /**
+     * Set point size
+     * @param pointSize the point size
+     */
+    public void setPointSize(double pointSize) {
+        this.pointSize.set(pointSize);
+    }
+
+    /**
+     * Set the X axis title.
+     * @param value The axis label
+     */
+    public void xLabel(String value) {
+        this.xLabel.set(value);
+    }
+
+    /**
+     * Set the Y axis title
+     * @param value The axis label
+     */
+    public void yLabel(String value) {
+        this.yLabel.set(value);
+    }
+
+    private void resampleAndUpdate() {
+        Collections.shuffle(data, new Random(rngSeed.get()));
+        series.getFirst().getData().clear();
+        series.getFirst().getData().setAll(data.subList(0, Math.min(maxPoints.get(), data.size())));
+        updateChart();
+    }
+
+    void setDataFromMeasurements(Collection<PathObject> pathObjects, String xMeas, String yMeas) {
```

Comment:
Use `<? extends PathObject>` whenever possible, to avoid unnecessary restriction.

Reply:
Would also prefer full words instead of `xMeas` and `yMeas` - not sure the extra cognitive autocompletion it requires is worth it.